### PR TITLE
qownnotes: move package to by-name directory

### PR DESCRIPTION
**Pull request by pbek**
Moved package to `by-name/qo/qownnotes/package.nix` and ran `nixfmt` on it.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pbek** [comment]
@NixOS/nixpkgs-merge-bot merge

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -4951,7 +4951,7 @@ with pkgs;
 
   qosmic = libsForQt5.callPackage ../applications/graphics/qosmic { };
 
-  qownnotes = qt6Packages.callPackage ../applications/office/qownnotes {
+  qownnotes = callPackage ../by-name/qo/qownnotes/package.nix {
```
Comment (lucasew) [inline pkgs/top-level/all-packages.nix:4954]: AFAIK this should be removed when moving to by-name

**Comment by lucasew** [comment]
> > pkgs/top-level/all-packages.nix is not in pkgs/by-name/
> 
> @wineee, could you help me out? Can this be mitigated? I can't remove the entry from `pkgs/top-level/all-packages.nix` because of the `stdenv` override, can't I?
> 
> ```nix
>   qownnotes = callPackage ../by-name/qo/qownnotes/package.nix {
>     stdenv = if stdenv.hostPlatform.isDarwin then darwin.apple_sdk_11_0.stdenv else stdenv;
>   };
> ```

You can. Just move the check to the package itself.

**Comment by wineee** [comment]
`darwin.apple_sdk_11_0.stdenv` should be same with `stdenv` now:
https://github.com/NixOS/nixpkgs/pull/352129

**Comment by lucasew** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 369466`

---
### `x86_64-linux`

**Comment by pbek** [comment]
> @NixOS/nixpkgs-merge-bot merge
> > @pbek merge not permitted (https://github.com/NixOS/nixpkgs/issues/305350):
> > pkgs/top-level/all-packages.nix is not in pkgs/by-name/

Ah, the error message came just because I'm not permitted to merge myself (as maintainer of qownnotes), because I had to touch `pkgs/top-level/all-packages.nix`. All is good. Silly me, just realized when I woke up today. 😊

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (pbek) [reply pkgs/top-level/all-packages.nix:4954]: I thought those declarations should still be done there if needed (see https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#changing-implicit-attribute-defaults). I saw many by-name packages in `pkgs/top-level/all-packages.nix`. 🤔

**Comment by pbek** [comment]
> darwin.apple_sdk_11_0.stdenv should be same with stdenv now:
> https://github.com/NixOS/nixpkgs/pull/352129

@wineee, so you think I can remove the lines in `pkgs/top-level/all-packages.nix` altogether?

**Comment by wineee** [comment]
> > darwin.apple_sdk_11_0.stdenv should be same with stdenv now:
> > #352129
> 
> @wineee, so you think I can and should remove the lines in `pkgs/top-level/all-packages.nix` altogether?

Yes, this part of the code should not be needed

**Comment by pbek** [comment]
> Yes, this part of the code should not be needed

Great, thank you! I removed the entry! Builds still under Linux.

**Comment by pbek** [comment]
Maybe the script `Fix list numbers` can help you.

**Pull request merged**